Source/WebCore/PAL/pal/text/TextCodec.cpp

```diff
@@ -49,14 +49,20 @@ int TextCodec::getUnencodableReplacement(char32_t codePoint, UnencodableHandling
         codePoint = replacementCharacter;
 
     switch (handling) {
-    case UnencodableHandling::Entities:
-        return snprintf(replacement.data(), sizeof(UnencodableReplacementArray), "&#%u;", static_cast<unsigned>(codePoint));
-    case UnencodableHandling::URLEncodedEntities:
-        return snprintf(replacement.data(), sizeof(UnencodableReplacementArray), "%%26%%23%u%%3B", static_cast<unsigned>(codePoint));
+    case UnencodableHandling::Entities: {
+        int count = snprintf(replacement.data(), sizeof(UnencodableReplacementArray), "&#%u;", static_cast<unsigned>(codePoint));
+        ASSERT(count >= 0);
+        return std::span { replacement }.first(static_cast<size_t>(std::max<int>(0, count)));
     }
+    case UnencodableHandling::URLEncodedEntities: {
+        int count = snprintf(replacement.data(), sizeof(UnencodableReplacementArray), "%%26%%23%u%%3B", static_cast<unsigned>(codePoint));
+        ASSERT(count >= 0);
+        return std::span { replacement }.first(static_cast<size_t>(std::max<int>(0, count)));
```

Comment:
Is there some way to make this less repetitive? Maybe a function that wraps `snprintf` that we can use twice? It's notable that the only difference is the format string.

Reply:
Can't use a lambda because clang doesn't allow a printf format attribute to be applied to a lambda, and then the `-Wformat-nonliteral` warning can't be fixed:

```
    __attribute__((__format__(__printf__, 2, 3)))
    auto formatUnencodableReplacement = [&] (const char* const formatString, unsigned codePoint) -> std::span<char> {
        int count = snprintf(replacement.data(), sizeof(UnencodableReplacementArray), formatString, codePoint);
        ASSERT(count >= 0);
        return std::span { replacement }.first(std::max<int>(0, count));
    };
```

```
Source/WebCore/PAL/pal/text/TextCodec.cpp:51:20: error: '__format__' attribute only applies to Objective-C methods, blocks, and non-K&R-style functions [-Werror,-Wignored-attributes]
    __attribute__((__format__(__printf__, 2, 3)))
                   ^
Source/WebCore/PAL/pal/text/TextCodec.cpp:53:85: error: format string is not a string literal [-Werror,-Wformat-nonliteral]
      int count = snprintf(replacement.data(), sizeof(UnencodableReplacementArray), formatString, codePoint);
                                                                                    ^~~~~~~~~~~~
2 errors generated.
```

So I guess it has to be a static function.

Reply:
In order to extract a function, I need to make it variadic.  I don't think it's worth it for two `snprintf()` function calls.

Reply:
OK I understand.

Reply:
[Adding this comment after Darin resolved the issue because it was "stuck" as a pending review comment.]

I feel this is much less readable than just leaving the `snprintf()` calls where they were:

```
__attribute__((__format__(__printf__, 2, 3)))
static std::span<char> formatUnencodableReplacement(UnencodableReplacementArray& replacement, const char* const format, ...)
{
    va_list ap;
    va_start(ap, format);
ALLOW_NONLITERAL_FORMAT_BEGIN
    int count = snprintf(replacement.data(), sizeof(UnencodableReplacementArray), format, ap);
ALLOW_NONLITERAL_FORMAT_END
    va_end(ap);
    ASSERT(count >= 0);
    return std::span { replacement }.first(std::max<int>(0, count));
}
```

Reply:
Yes I agree.